### ISSUE TITLE
Apply Configs Will Not Save.

### ISSUE THREAD

**Issue by JAP42** [tiredofit/docker-freepbx#64]
I am trying to set this up on a Ubuntu Based VPS. On a new install I immediately get a "Apply Config" button. It pops the "Restarting" Dialog up and then goes nowhere. I left that going for about 20 mins at one point. I can referesh the page and the settings seem to have applied. But nothing actually works.

1. I tried recreating the containers
2. I tried Deleting all the data folders and recreating. 
3. I tried Deleting all the data and removing the images to re-pull. 

All three had the same result. Can not apply settings. 

	version: '2'

	services:
	  freepbx-app:
		container_name: freepbx-app
		image: tiredofit/freepbx
		ports:
		 #### If you aren't using a reverse proxy
		  - 8002:80
		 #### If you want SSL Support and not using a reverse proxy
		 #- 443:443
		  - 5060:5060
		  - 5160:5160
		  - 18000-18100:18000-18100/udp
		 #### Flash Operator Panel
		  - 4445:4445
		volumes:
		  - ./certs:/certs
		  - ./data:/data
		  - ./logs:/var/log
		  - ./data/www:/var/www/html
		 ### Only Enable this option below if you set DB_EMBEDDED=TRUE
		 #- ./db:/var/lib/mysql
		 ### You can drop custom files overtop of the image if you have made modifications to modules/css/whatever - Use with care
		 #- ./assets/custom:/assets/custom

		environment:
		  - VIRTUAL_HOST=voice.####.com
		  - VIRTUAL_NETWORK=nginx-proxy
		 ### If you want to connect to the SSL Enabled Container
		 #- VIRTUAL_PORT=443
		 #- VIRTUAL_PROTO=https
		  - VIRTUAL_PORT=80
		  - LETSENCRYPT_HOST=voice.####.com
		  - LETSENCRYPT_EMAIL=####

		  - ZABBIX_HOSTNAME=freepbx-app

		  - RTP_START=18000
		  - RTP_FINISH=18100

		 ## Use for External MySQL Server
		  - DB_EMBEDDED=FALSE

		 ### These are only necessary if DB_EMBEDDED=FALSE
		  - DB_HOST=freepbx-db
		  - DB_PORT=3306
		  - DB_NAME=asterisk
		  - DB_USER=asterisk
		  - DB_PASS=####

		 ### If you are using TLS Support for Apache to listen on 443 in the container drop them in /certs and set these:
		 #- TLS_CERT=cert.pem
		 #- TLS_KEY=key.pem

		restart: always
		networks:
		  - proxy-tier

		### These final lines are for Fail2ban. If you don't want, comment and also add ENABLE_FAIL2BAN=FALSE to your environment
		cap_add:
		  - NET_ADMIN
		privileged: true

	  freepbx-db:
		container_name: freepbx-db
		image: tiredofit/mariadb
		restart: always
		volumes:
		  - ./db:/var/lib/mysql
		environment:
		  - MYSQL_ROOT_PASSWORD=password
		  - MYSQL_DATABASE=asterisk
		  - MYSQL_USER=asterisk
		  - MYSQL_PASSWORD=####
		networks:
		  - proxy-tier

	  freepbx-db-backup:
		container_name: freepbx-db-backup
		image: tiredofit/db-backup
		links:
		 - freepbx-db
		volumes:
		  - ./dbbackup:/backup
		environment:
		  - ZABBIX_HOSTNAME=freepbx-db-backup
		  - DB_HOST=freepbx-db
		  - DB_TYPE=mariadb
		  - DB_NAME=asterisk
		  - DB_USER=asterisk
		  - DB_PASS=####
		  - DB_DUMP_FREQ=1440
		  - DB_DUMP_BEGIN=0000
		  - DB_CLEANUP_TIME=8640
		  - COMPRESSION=BZ
		  - MD5=TRUE
		networks:
		  - proxy-tier
		restart: always

	networks:
	  proxy-tier:
		external:
		  name: ee-global-frontend-network

**Comment by barhom** [comment]
#65 same issue?

**Comment by JAP42** [comment]
Nope, I did try your suggestion there. My apply button works, the UI just never resets. Although so far about 1 in 3 installs will actually activate the rest of the modules when I download and install them. There is next to no logging in this. Asterisk is going like crazy but the UI does not seem to log anything. Ether internally or to the browser. Is there a specific host config that is known to work? Like I said, I am using Ubuntu. I dont see a reference to what tiredofit is using.

**Comment by tiredofit** [comment]
Alot of times we have been seeing these upstream errors from Freepbx - I believe I am going to need to switch to a different method and start pinning stable releases to this image as oposed to just pulling from their git repositories.
I am moving and will be away for a week, once I return I will look into this.

@JAP42 I am using Arch Linux with kernel 5.0, btrfs file systems.

**Comment by JAP42** [comment]
Yay, Arch, the one linux flavor I have absolutly no experience with. lol. 

I just tried V14 and although I can apply configs, modules will not properly activate just like 15. I would deffinaly see about a stable branch pulling from a known stable freepbx build. I ended up using fusionpbx because I could not get freepbx working in any senario. Except their own OS which I would prefer to consolidate my services to 1 server which rules that out. 

Good Luck on the move!

**Comment by tiredofit** [comment]
Hi sorry about the delay, as per the other issue #65 I cannot recreate with a fresh install. This is frustrating.

**Comment by thegitcat** [comment]
I had the same issue. I am new here so I am not sure how to help, but I went on the freepbx source code and replace the code from this file:

https://raw.githubusercontent.com/FreePBX/framework/release/15.0/amp_conf/htdocs/admin/libraries/BMO/Framework.class.php 

to the file in container.
